commit 8a2ce0555dc4e6264d8e6c1bb8a7f84921c9e79d (refs/changes/49/2849/4)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-11-06T19:39:31-08:00 (1 year, 11 months ago)
    
    VOLAPI-86 store role of NFS volumes storage VMs in internal_metadata instead of tags

diff --git a/lib/endpoints/volumes.js b/lib/endpoints/volumes.js
index 6420dd7..aba4412 100644
--- a/lib/endpoints/volumes.js
+++ b/lib/endpoints/volumes.js
@@ -94,8 +94,8 @@ function _buildStorageVMPayload(volumeParams, storageVmUuid, imageUuid,
         // VM is lost.
         delegate_dataset: true,
         owner_uuid: volumeParams.owner_uuid,
-        tags: {
-            smartdc_role: volumeUtils.NFS_SHARED_VOLUME_SMARTDC_ROLE
+        internal_metadata: {
+            'sdc:system_role': volumeUtils.NFS_SHARED_VOLUME_SYSTEM_ROLE
         }
     };
 
diff --git a/lib/volumes.js b/lib/volumes.js
index 2b7a246..956fdbd 100644
--- a/lib/volumes.js
+++ b/lib/volumes.js
@@ -13,11 +13,11 @@ var assert = require('assert-plus');
 var NFS_SHARED_VOLUME_EXPORTS_BASEDIR = '/exports';
 var NFS_SHARED_VOLUME_EXPORTS_DIRNAME = 'data';
 var NFS_SHARED_VOLUME_VM_ALIAS_PREFIX = 'nfs-shared-volume';
-var NFS_SHARED_VOLUME_SMARTDC_ROLE = 'nfsvolumestorage';
+var NFS_SHARED_VOLUME_SYSTEM_ROLE = 'nfsvolumestorage';
 
 module.exports = {
     NFS_SHARED_VOLUME_EXPORTS_BASEDIR: NFS_SHARED_VOLUME_EXPORTS_BASEDIR,
     NFS_SHARED_VOLUME_EXPORTS_DIRNAME: NFS_SHARED_VOLUME_EXPORTS_DIRNAME,
-    NFS_SHARED_VOLUME_SMARTDC_ROLE: NFS_SHARED_VOLUME_SMARTDC_ROLE,
+    NFS_SHARED_VOLUME_SYSTEM_ROLE: NFS_SHARED_VOLUME_SYSTEM_ROLE,
     NFS_SHARED_VOLUME_VM_ALIAS_PREFIX: NFS_SHARED_VOLUME_VM_ALIAS_PREFIX
 };
\ No newline at end of file
diff --git a/volapi-updater.js b/volapi-updater.js
index 1fcf3e8..d11197e 100644
--- a/volapi-updater.js
+++ b/volapi-updater.js
@@ -554,7 +554,8 @@ function updateAllVolumesFromVmApi(vmapiClient, log, callback) {
     mod_assert.func(callback, 'callback');
 
     vmapiClient.listVms({
-        'tag.smartdc_role': mod_volumeUtils.NFS_SHARED_VOLUME_SMARTDC_ROLE
+        'internal_metadata.sdc:system_role':
+            mod_volumeUtils.NFS_SHARED_VOLUME_SYSTEM_ROLE
     }, function onGetAllVolumesVms(getVmsErr, volumeVms) {
         if (getVmsErr) {
             log.error({error: getVmsErr}, 'Error when fetching VMs');
